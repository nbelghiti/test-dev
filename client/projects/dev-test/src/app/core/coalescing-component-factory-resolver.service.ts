/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
