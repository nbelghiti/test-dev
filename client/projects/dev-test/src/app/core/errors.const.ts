export const ToastrConfig = {
	maxOpened: 2,
	timeOut: 8000
};
