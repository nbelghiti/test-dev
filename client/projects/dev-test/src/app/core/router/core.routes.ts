import { Routes } from '@angular/router';

import { AuthGuard, CallbackComponent, LoginComponent } from 'adb-auth';

import { ErrorPage } from '~/error/pages/error.page';

export const CORE_ROUTES: Routes = [{
	path: 'callback',
	component: CallbackComponent,
}, {
	path: 'login',
	redirectTo: 'projects',
	pathMatch: 'full',
}, {
	path: '',
	redirectTo: 'projects',
	pathMatch: 'full',
}, {
	path: 'articles',
	loadChildren: () => import('~/articles/articles.module').then((mod) => mod.ArticlesModule),
}, {
	path: 'projects',
	loadChildren: () => import('~/projects/projects.module').then((mod) => mod.ProjectsModule),
}, {
	path: 'not-found',
	component: ErrorPage
}, {
	path: '**',
	redirectTo: 'not-found',
}];
