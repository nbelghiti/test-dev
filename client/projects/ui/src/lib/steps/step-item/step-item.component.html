<div
	class="c-step-item"
	[ngClass]="{
		'c-step-item--completed': state === StepType.completed,
		'c-step-item--active': state === StepType.active
	}"
>
	<span class="c-step-item__number">
		{{ number }}
	</span>

	<div class="c-step-item__content">
		<p class="c-step-item__title">
			{{ title }}
		</p>

		<p class="c-step-item__description">
			{{ description }}
		</p>
	</div>
</div>
