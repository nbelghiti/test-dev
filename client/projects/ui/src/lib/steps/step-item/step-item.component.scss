// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$step-number-size: 6rem;

// =============================================================================
// :: Step Item
// =============================================================================
.c-step-item {
	display: flex;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-step-item__number {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $step-number-size;
	height: $step-number-size;

	font-family: $font-family-tertiary;
	font-size: $font-size-4x-large;
	font-weight: $font-weight-extrablack;
	font-variant-numeric: tabular-nums lining-nums;
	line-height: $step-number-size;
	text-align: center;

	color: $text-color-grey-76;
	border-radius: 50%;
	background-color: $background-color-base;
}

.c-step-item__content {
	max-width: 17.5rem;
	margin-left: 1.5rem;
}

.c-step-item__title {
	font-size: $font-size-base;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;

	color: $text-color-grey-56;
}

.c-step-item__description {
	font-size: $font-size-1x-small;
	line-height: $line-height-base;

	color: $text-color-grey-56;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-step-item--active {

	.c-step-item__number {
		color: $text-color-white;
		border: none;
		background-color: $primary-color;
	}

	.c-step-item__title {
		color: $text-color-base;
	}

	.c-step-item__description {
		color: $text-color-grey-46;
	}
}

.c-step-item--completed {

	.c-step-item__number {
		color: $text-color-white;
		border: none;
		background-color: $background-color-orange-78;
	}
}
