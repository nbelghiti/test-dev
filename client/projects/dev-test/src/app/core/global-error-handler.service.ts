import { ErrorHandler, Inject, Injectable, Injector } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';
import { ToastrService } from 'ngx-toastr';

@Injectable()
export class GlobalErrorHandler extends ErrorHandler {
	constructor(
		@Inject(Injector) private readonly injector: Injector,
		private translate: TranslateService
	) {
		super();
	}

	handleError(error) {
		this.translate.get('ERRORS__GENERAL').subscribe((message: string) => {
			this.toastr.error(message, null, { onActivateTick: true });
		});

		console.error(error);

		super.handleError(error);
	}

	/**
	 * Need to get ToastrService from injector rather than constructor injection to avoid cyclic dependency error
	 * https://github.com/scttcper/ngx-toastr/issues/327
	 */
	private get toastr(): ToastrService {
		return this.injector.get(ToastrService);
	}
}
