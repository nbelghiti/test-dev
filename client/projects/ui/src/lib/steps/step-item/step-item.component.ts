import { Component, Input } from '@angular/core';

import { StepType } from '../steps.types';

@Component({
	selector: 'ui-step-item',
	templateUrl: './step-item.component.html',
	styleUrls: ['./step-item.component.scss']
})

export class StepItemComponent {
	@Input() public number: number;
	@Input() public state: StepType = StepType.default;
	@Input() public title: string;
	@Input() public description: string;
	public StepType = StepType;
}
