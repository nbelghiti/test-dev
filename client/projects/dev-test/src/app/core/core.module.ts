import { ErrorHandler, NgModule } from '@angular/core';

import { CoalescingComponentFactoryResolver } from './coalescing-component-factory-resolver.service';
import { ContextModule } from './context/context.module';
import { GlobalErrorHandler } from './global-error-handler.service';
import { RouterModule } from './router/router.module';
import { UserModule } from './user/user.module';

@NgModule({
	imports: [
		ContextModule,
		RouterModule,
		UserModule,
	],
	providers: [
		CoalescingComponentFactoryResolver,
		{
			provide: ErrorHandler,
			useClass: GlobalErrorHandler,
		},
	],
})
export class CoreModule {
	constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
		coalescingResolver.init();
	}
}
