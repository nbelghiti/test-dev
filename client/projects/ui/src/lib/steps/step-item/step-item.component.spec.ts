import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { StepItemComponent } from './step-item.component';

describe('StepItemComponent', () => {
	let component: StepItemComponent;
	let fixture: ComponentFixture<StepItemComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
			],
			declarations: [ StepItemComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(StepItemComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
