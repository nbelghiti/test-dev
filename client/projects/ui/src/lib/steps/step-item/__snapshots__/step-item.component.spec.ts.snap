// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepItemComponent Should render correctly 1`] = `
<ui-step-item
  StepType={[Function Object]}
  state=""
>
  <div
    class="c-step-item"
    ng-reflect-klass="c-step-item"
    ng-reflect-ng-class="[object Object]"
  >
    <span
      class="c-step-item__number"
    >
        
    </span>
    <div
      class="c-step-item__content"
    >
      <p
        class="c-step-item__title"
      >
          
      </p>
      <p
        class="c-step-item__description"
      >
          
      </p>
    </div>
  </div>
</ui-step-item>
`;
